llvm/lib/IR/Type.cpp

```diff
@@ -879,6 +879,10 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
         ScalableVectorType::get(Type::getInt8Ty(C), TotalNumElts));
   }
 
+  // DirectX intangible types
+  if (Name.starts_with("dx."))
```

Comment:
I'm surprised that we do a name lookup here, rather than key off some attributes, but I also see that this is how it is done for some spirv and riscv types.

Are we definitely ok with burning all the `dx.` prefixed names for this?

Reply:
Yeah, it's definitely kind of annoying to have this all be string based, but that's what it is.

The "intangible types" comment is maybe misleading here, but this isn't actually burning all of the `dx.` prefixed names. See the `spirv` case, where it further specializes based on the rest of the name.

Reply:
I have updated the comment.